http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/WsmDao.scala

```diff
@@ -0,0 +1,351 @@
+  }
+
+  implicit val wsmErrorReportDecoder: Decoder[WsmErrorReport] =
+    Decoder.forProduct3("message", "statusCode", "causes")(WsmErrorReport.apply)
+
+  implicit val deleteControlledAzureResourceResponseDecoder: Decoder[DeleteVmResult] = Decoder.instance { c =>
+    for {
+      jobReport <- c.downField("jobReport").as[WsmJobReport]
+      errorReport <- c.downField("errorReport").as[Option[WsmErrorReport]]
+    } yield DeleteVmResult(jobReport, errorReport)
+  }
+
+  implicit val createVmResultDecoder: Decoder[CreateVmResult] =
+    Decoder.forProduct3("azureVm", "jobReport", "errorReport")(CreateVmResult.apply)
+}
+
+object WsmEncoders {
+  implicit val controlledResourceIamRoleEncoder: Encoder[ControlledResourceIamRole] =
+    Encoder.encodeString.contramap(x => x.toString)
+  implicit val privateResourceUserEncoder: Encoder[PrivateResourceUser] =
+    Encoder.forProduct2("userName", "privateResourceIamRoles")(x => (x.userName.value, x.privateResourceIamRoles))
+  implicit val wsmCommonFieldsEncoder: Encoder[ControlledResourceCommonFields] =
+    Encoder.forProduct6("name",
+                        "description",
+                        "cloningInstructions",
+                        "accessScope",
+                        "managedBy",
+                        "privateResourceUser")(x =>
+      (x.name.value,
+       x.description.value,
+       x.cloningInstructions.toString,
+       x.accessScope.toString,
+       x.managedBy.toString,
+       x.privateResourceUser)
+    )
+
+  implicit val ipRequestDataEncoder: Encoder[CreateIpRequestData] =
+    Encoder.forProduct2("name", "region")(x => (x.name.value, x.region.toString))
+  implicit val createIpRequestEncoder: Encoder[CreateIpRequest] =
+    Encoder.forProduct2("common", "azureIp")(x => (x.common, x.ipData))
+
+  implicit val diskRequestDataEncoder: Encoder[CreateDiskRequestData] =
+    Encoder.forProduct3("name", "size", "region")(x => (x.name.value, x.size.gb, x.region.toString))
+  implicit val createDiskRequestEncoder: Encoder[CreateDiskRequest] =
+    Encoder.forProduct2("common", "azureDisk")(x => (x.common, x.diskData))
+
+  implicit val networkRequestDataEncoder: Encoder[CreateNetworkRequestData] =
+    Encoder.forProduct5("networkName", "subnetName", "addressSpaceCidr", "subnetAddressCidr", "region")(x =>
+      (x.networkName.value, x.subnetName.value, x.addressSpaceCidr.value, x.subnetAddressCidr.value, x.region.toString)
+    )
+  implicit val createNetworkRequestEncoder: Encoder[CreateNetworkRequest] =
+    Encoder.forProduct2("common", "azureNetwork")(x => (x.common, x.networkData))
+
+  implicit val vmRequestDataEncoder: Encoder[CreateVmRequestData] =
+    Encoder.forProduct7("name", "region", "vmSize", "vmImageUri", "ipId", "diskId", "networkId")(x =>
+      (x.name.asString,
+       x.region.toString,
+       x.vmSize.toString,
+       x.vmImageUri.imageUrl,
+       x.ipId.value.toString,
+       x.diskId.value.toString,
+       x.networkId.value.toString)
```

Comment:
if you have encoder for these value classes, you shouldn't need to call `toString`..encoders for value classes should be in https://github.com/DataBiosphere/leonardo/blob/develop/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodec.scala

Reply:
I'm a bit wary to move them to core, as I see WSM DAO encoders/decoders as mainly a back-leo implementation details;  I don't see us hitting WSM directly in any integration tests. These are not the request models that automation tests will use when they hit the front-end service.

We can always move them if its relevant